### PR TITLE
Ids catalogue properties #515

### LINKED ISSUE #515: ids for catalogue item properties

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

- Change to use id in the post and patch request for items and catalogue items 
- Change it so it can only edit the name
-  

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes #515

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -82,7 +82,7 @@ jobs:
           MONGO_INITDB_ROOT_PASSWORD: example
 
       ims-api:
-        image: harbor.stfc.ac.uk/inventory-management-system/ims-api:develop
+        image: harbor.stfc.ac.uk/inventory-management-system/ims-api:pr-240
```
Comment (joshuadkitenge) [inline .github/workflows/ci-build.yml:85]:
```suggestion
        image: harbor.stfc.ac.uk/inventory-management-system/ims-api:develop
```

File: src/mocks/handlers.ts [hunk 2]
```diff
@@ -65,7 +66,12 @@ export const handlers = [
         parent_id: null,
       };
     }
-
+    body = {
+      ...body,
+      catalogue_item_properties: body.catalogue_item_properties?.map(
+        (property) => ({ ...property, id: generateUniqueId('test_id') })
```
Comment (joelvdavies) [inline src/mocks/handlers.ts:72]:
```suggestion
        (property) => ({ ...property, id: generateUniqueId('test_id_') })
```
For consistency with the other uses of `generateUniqueId`.

**Review by joelvdavies — COMMENTED** [review]
Just a couple minor comments but LGTM.

File: src/api/catalogueCategories.test.tsx [hunk 3]
```diff
@@ -70,41 +70,47 @@ describe('catalogue category api functions', () => {
       expect(result.current.data).toEqual({
```
[the remark this answers is not in the document]
Comment (joelvdavies) [reply src/api/catalogueCategories.test.tsx:70]: :smile: I meant the `posts a request to add a user session and returns successful response` it doesn't show up here.

**Review by MatteoGuarnaccia5 — DISMISSED** [review]
LGTM

**Comment by joshuadkitenge** [comment]
the e2e tests are failing because the backend hasn't been merged in

**Review by joelvdavies — DISMISSED** [review]

**Pull request merged**